docs/contributing/architecture/rest-api2

```diff
@@ -0,0 +1,555 @@
+Viewset with ``importers`` as the ``Importer.endpoint_name``. The
+``FileImporterViewset.endpoint_name`` is ``file``, so requests to ``api/v3/importers/file/*`` are
+handled by :class`TODO(asmacdo)~pulp_file.app.viewsets.FileImporterViewset`. For a more detailed explanation of
+how "Master/Detail" works, see :doc:`../contributing/architecture/rest-api`.
+
+FileImporterViewset defines a "Detail Route" called ``sync``, which accepts POST requests to
+to the endpoint in the example. The ``sync`` route is primarily responsible for deploying a Celery
+Task. In this case, it deploys the :function`TODO(asmacdo)~pulp_file.app.tasks.sync`.
+
+The response is a 202 and contains an href to view the Task. For more on
+how Tasks work, see :doc`TODO(asmacdo, stub out Tasks)`.
+
+.. code::
+
+    [
+        {
+            "_href": "http://pulp3.dev:8000/api/v3/tasks/438bc62e-2431-42c7-a85f-ddc05ad7b039/",
+            "task_id": "438bc62e-2431-42c7-a85f-ddc05ad7b039"
+        }
+    ]
+
+
+Fundamentals
+------------
+
+The Pulp 3 API is intended to be decoupled from the data model, which the Django REST Framework
+(DRF) makes it pretty easy to do. Where needed, support classes have been added to help the REST
+API correctly and consistently represent our specialized models and relationships (most notably
+the Master/Detail relationship).
+
+Our API starts at a :term:`DRF` :term:`Router`. Each :term:`ViewSet` is attached to this
+router, and the router's routes are exposed in urls.py. Subclasses of
+:class:`pulp.app.viewsets.base.NamedModelViewSet` are automatically registered with the API router,
+and all ViewSets created by plugins should be subclasses of this base class. NamedModelViewSets are
+associated with a Django queryset, a :term:`Serializer` that is able to represent members of the
+Django queryset in the API, and an endpoint name used when registering
+the ViewSet with the API router.
+
+All models exposed via the API must have a corresponding Serializer. Each NamedModelViewSet must
+be related to a serializer and a queryset of model instances to serialize.
+
+Since Serializers and ViewSets are so closely related to the models they represent, the
+serializers and viewsets directories are laid out similarly to the models directory to help keep
+things consistent and easy to find.
+
+The API basic component tree looks like this::
+
+    router
+    |-- viewset
+    |   |-- queryset
+    |   |-- serializer
+    |
+    |-- viewset
+    |   |-- ...
+    |
+    |-- ...
+
+When creating API components, consider these guidelines:
+
+* Where possible, API components representing models will be defined in files whose names match
+  the corresponding file names of the models represented. For example, if you're defining the
+  serializer for a model found in pulp.app.models.consumer, the serializer should be defined in
+  pulp.app.serializers.consumer, and imported by name into pulp.app.serializers.
```

Comment:
Is this the best example?  I thought we weren't going to have consumer stuff in Pulp anymore.